src/main/java/cs2103/v15_1j/jimjim/JJMain.java

```diff
@@ -31,12 +34,13 @@ public void start(Stage primaryStage) throws Exception {
 		Storage storage = new JJStorage();
 		Parser parser = new JJParser();
 		Searcher searcher = new JJSearcher();
+		Stack<Command> undoCommandHistory = new Stack<Command>();
```

Comment:
How will you implement redo? One way I can think of is to have another stack for the things that you pop out from `undoCommandHistory`.